### DISCUSSION TITLE
Getting Started with the Replicad App Example

### DISCUSSION THREAD

**Discussion by BarbourSmith** [sgenoud/replicad#68] [Q&A] [answered]
Hello! I'm a huge fan of this project and I want to get started with using it. The visualizer works great and the documentation for how to use it is excellent.

The next step is that I would like to try to run the [replicad app example](https://github.com/sgenoud/replicad/tree/main/packages/replicad-app-example) locally, but I'm not quite where to begin. Forgive me if it's obvious, I don't know much about react.

My first guess was to clone the repo and run `npm install` probably followed by `prepare-packages` but I'm running into an issue with some of the node dependencies being unable to resolve their version numbers.

**What are the steps to getting the basic example running locally?**

**Is it important that I use react? If it is important that I use react with Replicad I can learn it and switch over from vanilla JS.**

**Comment by BarbourSmith** [comment]
Upon further investigation I see that I probably should have been running `npm install` and then `npm start` from within _replicad-app-example_ ...but doing that I am seeing the same issue with typedoc version conflicts 😛

**Comment by sgenoud** [comment, marked as answer]
I do not advice to just clone the multirepo and then run from packages (it needs some magic with the multi repo logic, this is not great).

What I would advice you do to is copy the sample app out of the multi repo and then install and run. So something in that spirit:

```sh
git clone git@github.com:sgenoud/replicad.git
cp -r replicad/packages/replicad-app-example .
cd replicad-app-example
npm install
npm start
```

Note that I have updated the dependencies while I was testing, so pull within the repo before copying the demo app!

**Comment by sgenoud** [comment]
> Is it important that I use react? If it is important that I use react with Replicad I can learn it and switch over from vanilla JS.

It is not important to use react. This is just a way for me to make it easy to use threejs (via [react three fiber](https://docs.pmnd.rs/react-three-fiber/getting-started/introduction)). But in theory you can use vanilla JS.

**Comment by sgenoud** [comment]
And just a question on my side - what do you want to do with the sample app? Depending on your use case I can give you some advice on where to look to start developing your app!

**Comment by BarbourSmith** [comment]
Sorry about the slow response, I was camping :) 

I'm working on this project: https://github.com/MaslowCNC/Maslow-Create

It's a flow based 3D modeling program with the option to support writing code directly for more complex things. The goal is to make it easier for folks to share designs for CNC routers. 

I did a little research on React and it seems pretty nice so I think I'll try to make everything work in React. I should be able to dig into it a bit next week. 

Again, super cool project 👍 👍

**Comment by BarbourSmith** [comment]
Life got in the way and it took me longer to get to this than I expected, but I just had a chance to play with it and the example is working great. Thank you 😀

**Reply by BarbourSmith** [reply]
Just out of curiosity why is it important to do the copy out of the main repo? I wouldn't expect that to matter, but it does

**Reply by sgenoud** [reply]
It is a monorepo which uses the code from the library itself (and not the npm version). So it makes the whole build process a little bit more complicated.

**Reply by BarbourSmith** [reply]
That makes sense. Thanks!
